backend/repositories/match-repository.js

```diff
@@ -79,5 +80,6 @@ const constructMatch = async (matchDescription) => {
   */
 exports.recordMatch = async (matchDescription) => {
     const match = await constructMatch(matchDescription)
+    bot.postResultToSlack(match);
```

Comment:
Wait until the match is stored (there can be exceptions thrown) and post afterwards.